src/installer/install_utils.py

```diff
@@ -0,0 +1,368 @@
+import getpass
+import os
+import platform
+import subprocess
+import sys
+from dataclasses import dataclass
+from typing import Any, List, Optional, Union
+
+import distro
+
+from installer.constants import (
+    ARCH_MAPPING,
+    GPG_KEYS,
+    INSTALL_COMMANDS,
+    KEYRING_PATH,
+    REPO_PATHS,
+    REPO_STRINGS,
+    REQUIRED_DOCKER_PKGS,
+    REQUIRED_PKGS,
+    REQUIRED_PYTHON_PKGS,
+    UPDATE_COMMANDS,
+)
+
+# Color constants (ANSI color codes)
+RED = "\033[0;31m"
```

Comment:
please, let's avoid this bash like style, let's rely on `util/util.py` that is using `rich.console`.
If you need specific printing function add those there (and use the functionalities of `rich.console`)